src/status_im2/contexts/chat/messages/drawers/view.cljs

```diff
@@ -0,0 +1,144 @@
+       :id       :delete-for-all}])))
+
+(defn reactions [message-id]
+  [rn/view {:style {:flex-direction     :row
+                    :justify-content    :space-between
+                    :padding-horizontal 30
+                    :padding-top        5
+                    :padding-bottom     15}}
+   (doall
+    (for [[id icon] constants/reactions]
+      ;:let [active (own-reactions id)]]
+      ;;TODO reactions selector should be used https://www.figma.com/file/WQZcp6S0EnzxdTL4taoKDv/Design-System?node-id=9961%3A166549
+      ;; not implemented yet
+      ^{:key id}
+      [quo/button (merge
+                   {:size                40
+                    :type                :grey
+                    :icon                true
+                    :icon-no-color       true
+                    :accessibility-label (str "emoji-picker-" id)
+                    :on-press            #(do
+                                            (rf/dispatch [:models.reactions/send-emoji-reaction
+                                                          {:message-id message-id
+                                                           :emoji-id   id}])
+                                            (rf/dispatch [:bottom-sheet/hide]))})
+       ;(when active {:style {:background-color colors/neutral-10}}))
+       icon]))])
+
+;; TODO (flexsurfer) this probably should be a component in quo2 https://www.figma.com/file/WQZcp6S0EnzxdTL4taoKDv/Design-System?node-id=5626%3A158317&t=1JiRhBswpVXZ4SYF-3
+(defn reactions-and-actions [message-data]
+  (fn []
+    (let [actions (get-actions message-data)
+          main-actions (filter #(= (:type %) :main) actions)
+          danger-actions (filter #(= (:type %) :danger) actions)
+          admin-actions (filter #(= (:type %) :admin) actions)]
+      [:<>
+       ;; REACTIONS
+       [reactions (:message-id message-data)]
+
+       ;; MAIN ACTIONS
+       [rn/view {:style {:padding-horizontal 8}}
+        (for [action main-actions]
+          (let [on-press (:on-press action)]
+            ^{:key (:id action)}
+            [quo/menu-item
+             {:type                :main
+              :title               (:label action)
+              :accessibility-label (:label action)
+              :icon                (:icon action)
+              :on-press            #(do
+                                      (when on-press (on-press))
+                                      (rf/dispatch [:bottom-sheet/hide]))}]))
+        (when-not (empty? danger-actions)
+          [quo/separator])
+
+        ;; DANAGER ACTIONS
+        (for [action danger-actions]
+          (let [on-press (:on-press action)]
+            ^{:key (:id action)}
+            [quo/menu-item
+             {:type                :danger
+              :title               (:label action)
+              :accessibility-label (:label action)
+              :icon                (:icon action)
+              :on-press            #(do
```

Comment:
[Style guide] https://guide.clojure.style/#no-multiple-forms-fn-literals.

FYI: There are other instances of this styling issue.

Reply:
yeah i just moved this component, just didn't review it